src/main/java/com/bitso/examples/BitsoWebSocketExample.java

```diff
@@ -1,32 +1,212 @@
-    BitsoWebSocket ws;
+import org.json.JSONObject;
 
-    public BitsoWebSocketExample() {
-        Channels[] subscribeTo = { Channels.TRADES, Channels.DIFF_ORDERS, Channels.ORDERS };
-        ws = new BitsoWebSocket(subscribeTo);
-        ws.addObserver(this);
-        ws.connect();
-    }
+import com.bitso.Bitso;
+import com.bitso.BitsoBook;
+import com.bitso.BitsoOrder;
+import com.bitso.BitsoOrderBook;
+import com.bitso.BitsoOrderBook.PulicOrder;
+import com.bitso.helpers.Helpers;
+import com.bitso.websockets.BitsoChannels;
+import com.bitso.websockets.BitsoStreamDiffOrders;
+import com.bitso.websockets.BitsoStreamTrades;
+import com.bitso.websockets.BitsoWebSocket;
+import com.bitso.websockets.BitsoWebSocketObserver;
+import com.bitso.websockets.BitsoWebSocketPublicOrder;
 
-    @Override
+public class BitsoWebSocketExample extends BitsoWebSocketObserver{
+    
+    private OrderStructure mBids;
+    private OrderStructure mAsks;
+    
+    private BitsoOrderBook mLiveOrderBook;
+    private Bitso mBitso;
+        
     public void update(Observable o, Object arg) {
-        String message = arg.toString();
-        if (message.equals("disconnected")) {
-            System.out.println("The websocket got disconnected, reconnecting.");
-            ws.connect();
+        // Update message
+        if(arg instanceof String){
+            String messageReceived = ((String) arg);
+            JSONObject jsonObject = new JSONObject(messageReceived);
+            if(!jsonObject.has("action")){
+                String type = Helpers.getString(jsonObject, "type");
+                switch (type) {
+                    case "trades":
+                        BitsoStreamTrades trades = new BitsoStreamTrades(jsonObject);
+                        System.out.println(trades);
+                        break;
+                    case "diff-orders":
+                        BitsoStreamDiffOrders diff = new BitsoStreamDiffOrders(jsonObject);
+                        System.out.println(diff);
+                        mergeOrders(diff);
+                        System.out.println("Best ask: " + mAsks.getMinPrice());
+                        System.out.println("Best bid: " + mBids.getMaxPrice());
+                        break;
+                }
+            }
+        }
+        
+        // On connect/disconnect
+        if(arg instanceof Boolean){
+            mWSConnected = ((Boolean) arg);
+            if(mWSConnected){
+                System.out.println("Web socket is now connected");
+            }else{
+                System.out.println("Web socket is now disconnected");
+            }
+            
+        }
+    }
+    
+    public void mergeOrders(BitsoStreamDiffOrders diff){
+        OrderStructure orderStructure;
+        BitsoWebSocketPublicOrder[] bids = diff.getPayload();
+        for (BitsoWebSocketPublicOrder bitsoWebSocketPublicOrder : bids) {
+            if(bitsoWebSocketPublicOrder.getSide().compareTo(BitsoOrder.SIDE.BUY) == 0){
```

Comment:
Perhaps I'm just showing my poor java skills, but why can't this just be:
if(bitsoWebSocketPublicOrder.getSide() == BitsoOrder.SIDE.BUY)